app/tests/reader_studies_tests/test_views.py

```diff
@@ -465,6 +470,40 @@ def test_display_set_update(client):
     assert ds1.values.filter(pk=civ_json_file_new.pk).exists()
     assert ds1.values.get(interface=ci_json).value == {"foo": "new"}
 
+    # test new json file upload
+    upload = UserUploadFactory(filename="file.json", creator=user)
+    presigned_urls = upload.generate_presigned_urls(part_numbers=[1])
+    response = put(presigned_urls["1"], data=b'{"new": "content"}')
+    upload.complete_multipart_upload(
+        parts=[{"ETag": response.headers["ETag"], "PartNumber": 1}]
+    )
+    upload.save()
+
+    with django_capture_on_commit_callbacks(execute=True):
+        response = get_view_for_user(
+            viewname="reader-studies:display-set-update",
+            client=client,
+            reverse_kwargs={"pk": ds1.pk, "slug": rs.slug},
+            data={
+                ci_json.slug: '{"foo": "new"}',
+                ci_img.slug: str(im2.pk),
+                f"WidgetChoice-{ci_img.slug}": WidgetChoices.IMAGE_SEARCH.name,
+                ci_json_file.slug: str(upload.pk),
+                "order": 11,
+            },
+            user=user,
+            method=client.post,
+        )
+    assert response.status_code == 302
```

Comment:
It's a good idea to test which URL you are being directed to too as you could be say being pushed towards the login page.